docs/pages/linux-demo.mdx

```diff
@@ -44,38 +44,108 @@ If you want to get a feel for Teleport commands and capabilities without setting
+  enter the commands in this guide into an Amazon EC2 [user data
   script](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html),
   Google Compute Engine [startup
-  script](https://cloud.google.com/compute/docs/instances/startup-scripts),
-  or similar.
+  script](https://cloud.google.com/compute/docs/instances/startup-scripts), or
+  similar.
 
   For a quick demo environment you can use to follow this guide, consider
   installing our DigitalOcean 1-Click droplet. View the installation page on
   [DigitalOcean
   Marketplace](https://marketplace.digitalocean.com/apps/teleport). Once your
   droplet is ready, SSH into the droplet and follow the configuration wizard.
 
-- A multi-factor authenticator app such as [Authy](https://authy.com/download/),
-  [Google Authenticator](https://www.google.com/landing/2step/), or
-  [1Password](https://support.1password.com/one-time-passwords/).
-
 You must also have **one** of the following:
 - A registered domain name.
 - An authoritative DNS nameserver managed by your organization, plus an existing
   certificate authority. If using this approach, ensure that your browser is
   configured to use your organization's nameserver.
+</TabItem>
+<TabItem label="Local Docker container">
+
+1. Install [mkcert](https://github.com/FiloSottile/mkcert) so you can set up a
+   local certificate authority and create a certificate for running the Teleport
+   Web UI with HTTPS.
+
+1. Install the mkcert CA:
+
+   ```code
+   $ mkcert -install
+   ```
+
+1. Create a directory on your workstation in which to place TLS credentials for
+   Teleport:
+
+   ```code
+   $ mkdir teleport-tls
+   ```
+
+1. Generate a certificate and private key for Teleport:
+
+   ```code
+   $ cd teleport-tls
+   $ mkcert localhost
+   ```
+
+1. Add the mkcert CA certificate to the `teleport-tls` directory so your Docker
+   container can access it:
+
+   ```code
+   $ cp "$(mkcert -CAROOT)/rootCA.pem" .
+   ```
+
+1. Start a local Docker container where you can follow the remaining
+   instructions in this guide:
+
+   ```code
+   $ docker run -it -v .:/etc/teleport-tls -p 3030:443 ubuntu:22.04
+   ```
+
+1. Make sure `curl` is installed on your container:
+
+   ```code
+   $ apt-get update && apt-get install -y curl
+   ```
+
+1. On the container, move the mkcert CA certificate into the directory where
+   your container stores CA certs (installing `curl` sets this up for you). When
+   starting, Teleport verifies its TLS certificate against the CA:
+
+   ```code
+   $ cp /etc/teleport-tls/rootCA.pem /etc/ssl/certs/mkcertCA.pem
+   ```
+
+</TabItem>
+</Tabs>
+
+Finally, you will need a multi-factor authenticator app such as
+[Authy](https://authy.com/download/), [Google
+Authenticator](https://www.google.com/landing/2step/), or
+[1Password](https://support.1password.com/one-time-passwords/).
 
 ## Step 1/4. Configure DNS
 
 Teleport uses TLS to provide secure access to its Proxy Service and Auth
```

Comment:
It's a bit awkward that the first sentence under "Configure DNS" discusses TLS, not DNS.

Reply:
It doesn't seem like we need this paragraph, so I've removed it.